### PR TITLE
Drop `platform-version` support from `daml.yaml`

### PR DESCRIPTION
**Pull request by cocreature**
Doesn’t work properly with Canton sandbox, nobody uses it and product
agreed to deleting it.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Main.hs [hunk 1]
```diff
@@ -96,7 +91,6 @@ commandParser = subparser $ fold
     , command "deploy" (info (deployCmd <**> helper) deployCmdInfo)
     , command "ledger" (info (ledgerCmd <**> helper) ledgerCmdInfo)
     , command "run-jar" (info runJarCmd forwardOptions)
-    , command "run-platform-jar" (info runPlatformJarCmd forwardOptions)
```
Comment (sofiafaro-da) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Main.hs:99]: This command is used to run daml sandbox, navigator, json-api, etc so probably we need to keep it, just without special version logic.

**Review by sofiafaro-da — APPROVED** [review]
Thanks!

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Main.hs (hunk 1 above)
Comment (cocreature) [reply daml-assistant/daml-helper/src/DA/Daml/Helper/Main.hs:99]: Good catch, I’ve instead switched the sdk config to call `run-jar` which is identical except for the version logic.

File: release/sdk-config.yaml.tmpl [hunk 2]
```diff
@@ -47,11 +47,11 @@ commands:
 - name: sandbox
   path: daml-helper/daml-helper
   desc: "Launch Sandbox"
-  args: ["run-platform-jar", "--logback-config=sandbox-logback.xml", "sandbox"]
+  args: ["run-jar", "--logback-config=sandbox-logback.xml", "sandbox"]
 - name: sandbox-classic
   path: daml-helper/daml-helper
   desc: "Deprecated. Launch Sandbox Classic (the default Sandbox implementation for SDK <= 0.13.55)"
-  args: ["run-platform-jar", "--logback-config=sandbox-logback.xml", "sandbox-classic"]
+  args: ["run-jar", "--logback-config=sandbox-logback.xml", "sandbox-classic"]
```
Comment (sofiafaro-da) [inline release/sdk-config.yaml.tmpl:54]: Oh right, these need to go like,

```suggestion
  args: ["run-jar", "--logback-config=sandbox-logback.xml", "daml-sdk/daml-sdk.jar", "sandbox-classic"]
```

Comment (cocreature) [reply release/sdk-config.yaml.tmpl:54]: duh ofc they do sorry I’ll test this time before pushing :facepalm:

**Pull request merged**